source/OutfitterPanel.cpp

```diff
@@ -362,39 +444,30 @@ void OutfitterPanel::Buy(bool alreadyOwned)
 				player.Conditions()["license: " + licenseName] = true;
 	}
 
-	int modifier = Modifier();
-	for(int i = 0; i < modifier && CanBuy(alreadyOwned); ++i)
+	// Special case: maps.
+	int mapSize = selectedOutfit->Get("map");
+	if(mapSize)
 	{
-		// Special case: maps.
-		int mapSize = selectedOutfit->Get("map");
-		if(mapSize > 0)
-		{
-			if(!player.HasMapped(mapSize))
-			{
-				player.Map(mapSize);
-				int64_t price = player.StockDepreciation().Value(selectedOutfit, day);
-				player.Accounts().AddCredits(-price);
-			}
-			return;
-		}
+		player.Map(mapSize);
+		player.Accounts().AddCredits(-selectedOutfit->Cost());
+		return;
+	}
 
-		// Special case: licenses.
-		if(IsLicense(selectedOutfit->Name()))
-		{
-			auto &entry = player.Conditions()[LicenseName(selectedOutfit->Name())];
-			if(entry <= 0)
-			{
-				entry = true;
-				int64_t price = player.StockDepreciation().Value(selectedOutfit, day);
-				player.Accounts().AddCredits(-price);
-			}
-			return;
-		}
+	// Special case: licenses.
+	if(IsLicense(selectedOutfit->Name()))
+	{
+		player.Conditions()[LicenseName(selectedOutfit->Name())] = true;
+		player.Accounts().AddCredits(-selectedOutfit->Cost());
+		return;
+	}
 
```

Comment:
Restore the guards you removed from maps and licenses.

You also removed depreciation adjustments--why?

Reply:
The only place Buy is called is after CanBuy is called which already checks if the purchase is valid. It's at least redundant, (and perhaps as I stated elsewhere, a reason to further combine CanBuy and Buy)

Maps and license can't depreciate as far as I know.